lib/mapObjects/CObjectHandler.h

```diff
@@ -159,6 +159,7 @@ class DLL_LINKAGE CGObjectInstance : public IObjectInterface
 	bool visitableAt(int x, int y) const; //returns true if object is visitable at location (x, y) (h3m pos)
 	int3 visitablePos() const override;
 	int3 getPosition() const override;
+	int3 getTopVisiblePos() const;
```

Comment:
I had a hard time with `getTopVisiblePos()` and even harder to come with something more intelligible -> `getPositionForVisibleTop()` or `getPositionWhenTopIsVisible()`

Reply:
This gets the actual coordinates of the top visible tile of an object placed at its current position, and will always return correct result even if the object is completely outside the map.